pkg/datapath/linux/config.go

```diff
@@ -271,7 +271,9 @@ func (l *linuxDatapath) writeTemplateConfig(fw *bufio.Writer, e datapath.Endpoin
 			fmt.Fprint(fw, "#define ENABLE_ARP_RESPONDER 1\n")
 		}
 
-		fmt.Fprint(fw, "#define ENABLE_HOST_REDIRECT 1\n")
+		if option.Config.IPAM != option.IPAMENI {
+			fmt.Fprint(fw, "#define ENABLE_HOST_REDIRECT 1\n")
```

Comment:
Doesn't this break ipvlan and/or flannel?

Reply:
It only excludes this option for ENI mode. No change for IPVLAN or flannel mode.

Reply:
I read the other way around